rvip/include/rvip/pose_alignment.hpp

```diff
@@ -0,0 +1,185 @@
+
+                // Stage 3: Using ICP for fine-tuning of position
+                icp.setInputSource(sceneCloud);
+                icp.setInputTarget(localModelCloud);
+                // Setting ICP parameters
+                icp.setEuclideanFitnessEpsilon (0.000001);
+                icp.align(*alignCloud);
+                // Inverting transform to get tf frame from world origin to object position
+                icp_transform = icp.getFinalTransformation().inverse();
+                pcl::transformPointCloud (*localModelCloud, *finalTransformedCloud, icp_transform);
+
+                // Compiling all transforms made.
+                final_transform = icp_transform * super4pcs_transform * translate_to_obj_center;
+
+        }
+        catch(pcl:: PCLException& e) {
+                ROS_WARN("Unable to align. SUPER-4PCS and/or ICP failed.\n");
+                return;
+        }
+
+        static tf::TransformBroadcaster br;
+        tf::Vector3 origin2;
+        origin2.setValue(object.obj_x, object.obj_y, object.obj_z);
+        tf::Quaternion tfqt2 = straightenObjectZAxis(origin2, final_transform);
+
+        tf::Transform objTF2 = generateObjectTF(origin2, tfqt2);
+
+        br.sendTransform(tf::StampedTransform(objTF2, ros::Time::now(), object.frame_id, "T-T"));
+
+          tf::Vector3 origin;
+          origin.setValue(object.obj_x, object.obj_y, object.obj_z);
+
+          tf::Quaternion tfqt = straightenObjectZAxis(origin, final_transform);
+          tf::Transform objTF = generateObjectTF(origin, tfqt);
+
+          object.obj_tfs.push_back(objTF);
+          // DEBUG - Show marker of object with dynamic size, position and orientation.
+          visualization_msgs::Marker marker;
+          marker.header.frame_id = object.frame_id;
+          marker.header.stamp = ros::Time();
+          marker.type = visualization_msgs::Marker::CUBE;
+
+          marker.id = object.obj_markers.markers.size();
+          marker.pose.position.x = object.obj_x;
+          marker.pose.position.y = object.obj_y;
+          marker.pose.position.z = object.obj_z;
+          marker.pose.orientation.x = tfqt.x();
+          marker.pose.orientation.y = tfqt.y();
+          marker.pose.orientation.z = tfqt.z();
+          marker.pose.orientation.w = tfqt.w();
+          marker.scale.x = object.obj_length;
+          marker.scale.y = object.obj_width;
+          marker.scale.z = object.obj_height;
+          marker.color.r = 0.0f;
+          marker.color.g = 1.0f;
+          marker.color.b = 0.0f;
+          marker.color.a = 1.0;
+          marker.lifetime = ros::Duration(5.0);
+
+          object.obj_markers.markers.push_back(marker);
+
+          // DEBUG - Timestamp END - Print out time taken for pose alignment.
+          // end_ = ros::WallTime::now();
+          // double execution_time = (end_ - start_).toNSec() * 1e-6;
+          // ROS_INFO("Execution time for [Pose Alignment] (ms):  %f                  \r", execution_time);
+
+}
```

Comment:
This header does not appear to have any include guards.

Reply:
Addressed in [this commit](https://github.com/ros-industrial/rvip/pull/1/commits/d8d6f424b0cd678b617b8c1e1b15d1de4f998824).